### PR TITLE
v2: builder applyToRound

### PR DESCRIPTION
**Pull request by 0xKurt**
## Description

Flows affected and tested
- [x] Apply To Round on V1 QF
- [x] Apply To Round on V2 QF when both project and round are on same chain

NOTE: 
- CROSS CHAIN SYNCING PROJECT WILL BE HANDLED ON https://github.com/gitcoinco/grants-stack/issues/2928
- Applications will be shown only when https://github.com/gitcoinco/grants-stack/issues/2856 which will be the next ticket

Fixes: #2788

### CODE REVIEW

**Review by thelostone-mc — APPROVED** [review]

File: cache/solidity-files-cache.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"_format":"ethers-rs-sol-cache-3","paths":{"artifacts":"out","build_infos":"out/build-info","sources":"src","tests":"test","scripts":"script","libraries":["node_modules"]},"files":{}}
```
Comment (boudra) [inline cache/solidity-files-cache.json:1]: we should remove this file and add it to gitignore

**Review by boudra — APPROVED** [review]

File: packages/common/src/allo/roundIdentifier.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  | "DIRECT"
+  | "SUPERFLUID"
+  | "MICROGRANTS"
+  | "MICROGRANTSGOV"
+  | "MICROGRANTSHATS"
+  | "RFPSIMPLE"
+  | "RFPCOMMITTEE"
+  | "QVSIMPLE";
+
+export const getV2RoundType = (poolId: string): RoundType => {
```
Comment (boudra) [inline packages/common/src/allo/roundIdentifier.ts:12]: not sure if it helps but in the indexer we have the strategyName, including allo v1 qf and direct

**Review by boudra — APPROVED** [review]

File: packages/common/src/allo/roundIdentifier.ts (hunk 2 above)
Comment (boudra) [reply packages/common/src/allo/roundIdentifier.ts:12]: just want to make sure we won't have to repeat this code (for v2 and v1) if it's already in the indexer

**Review by michellema1208 — COMMENTED** [review]
hey @0xKurt / @codenamejason / @thelostone-mc - 

Confirming I was able to apply to a round on OP with a project on OP on v1! 

For v2, I'm pretty sure I can't test this until @boudra 's create round on v2 PR is live - is that correct? 

When I try to load an existing round's application in the staging v2 environment, this is what I see: 
Screen Shot 2024-02-14 at 11 52 29 AM

**Comment by 0xKurt** [comment]
> hey @0xKurt / @codenamejason / @thelostone-mc -
> 
> Confirming I was able to apply to a round on OP with a project on OP on v1!
> 
> For v2, I'm pretty sure I can't test this until @boudra 's create round on v2 PR is live - is that correct?
> 
> When I try to load an existing round's application in the staging v2 environment, this is what I see: Screen Shot 2024-02-14 at 11 52 29 AM


You may need to go to the frontend first and switch to v2.
The round should be open for the next 3 days: https://builder-staging-git-2788-builder-applytoround-grants-stack.vercel.app/#/chains/80001/rounds/2

**Comment by 0xKurt** [comment]
> @0xKurt / @codenamejason / @thelostone-mc -- I shared this on Discord but resharing here just in case: when I load the application link shared above, it redirects me to the Zuzalu round: https://builder-staging-git-2788-builder-applytoround-grants-stack.vercel.app/#/chains/80001/rounds/2 
> 
> I don't believe that's right? I've already switched to Allo v2 version as well. 
> 
> Screen Shot 2024-02-14 at 12 47 13 PM
> 

That's the right one 
We used it as test meta data

**Comment by michellema1208** [comment]
@0xKurt - oh gotcha, so this isn't the actual Zuzalu application, just a test copy with the same metadata. Just making sure I wasn't applying to the actual Zuzalu round / there was an indexing error where other applications on other chains with the same ID are showing up!

**Review by michellema1208 — APPROVED** [review]
confirming I'm able to test both of the following use cases: 

 Apply To Round on V1 QF
 Apply To Round on V2 QF when both project and round are on same chain

File: packages/common/src/allo/roundIdentifier.ts (hunk 2 above)
Comment (codenamejason) [reply packages/common/src/allo/roundIdentifier.ts:12]: Good call out, this is implemented as a short term fix.

**Pull request merged**